calico_kubernetes/calico_kubernetes.py

```diff
@@ -203,14 +212,24 @@ def _configure_interface(self):
         # so we allocate an IP which is already allocated to the node. We set
         # the subnet to /32 so that the routing table is not affected;
         # no traffic for the node_ip's subnet will use the /32 route.
+        logger.info('Adding IP %s to interface %s', node_ip, interface_name)
         check_call(['ip', 'addr', 'add', node_ip + '/32',
                     'dev', interface_name])
-        logger.info('Finished configuring network interface')
-        return ep
+
+        # Give Kubernetes a link to the endpoint
```

Comment:
We're not really giving Kubernetes a link to it, we're annotating the endpoint ID on the pod object so that the policy agent can look-up the corresponding endpoint for this pod.